mlir/lib/Target/LLVMIR/DebugImporter.cpp

```diff
@@ -85,10 +85,15 @@ DIDerivedTypeAttr DebugImporter::translateImpl(llvm::DIDerivedType *node) {
   DITypeAttr baseType = translate(node->getBaseType());
   if (node->getBaseType() && !baseType)
     return nullptr;
+  DINodeAttr extraData =
+      translate(llvm::dyn_cast_or_null<llvm::DINode>(node->getExtraData()));
```

Comment:
```suggestion
      translate(dyn_cast_if_present<llvm::DINode>(node->getExtraData()));
```
nit: I believe this is the more modern version.